src/pages/PaniniCreator.tsx

```diff
@@ -142,3 +147,89 @@ export default function PaniniCreator(props: PaniniCreatorProps) {
     </FormProvider>
   );
 }
+
+// possible bugs later on with generating due to HONEY_MUSTARD -> HONEY MUSTARD. Change later on if needed, also added pecorino.
+interface SandwichPayload {
+  sandwichName: string; // Max. 35 characters
+  cutlery: boolean;
+  napkins: boolean;
+  base: {
+    bread: "FULL GRAIN" | "WHEAT";
+    cheese: Array<"MOZZARELLA" | "STRACIATELLA" | "EDAM" | "GOUDA" | "PECORINO">;
+    meat: Array<"SALAMI" | "HAM" | "BACON" | "CHICKEN">;
+    dressing: Array<"OLIVE OIL" | "HONEY MUSTARD" | "RANCH" | "MAYO">;
+    vegetables: Array<
+      "SALAD" | "TOMATO" | "OBERGINE" | "BEETROOT" | "PICKLES" | "ONION" | "PEPPER" | "ASPARAGUS" | "CUCUMBER"
+    >;
+  };
+  extras: {
+    egg: Array<"FRIED EGG" | "OMELET" | "SCRAMBLED EGG">;
+    spreads: Array<"BUTTER" | "HUMMUS" | "GUACAMOLE">;
+    serving: "COLD" | "WARM" | "GRILLED";
+    topping: "SESAME" | null;
+  };
+}
+
+//for god known reasons ts won't accept this. cuz of some readonly stuff... that's why i'm using as "any".
+const literalMapper = function (val: string[]) {
+  return val.map((val) => z.literal(val)) as any;
+};
+
+const SandwichPayload = z.object({
+  sandwichName: z.string().min(1).max(35),
+  cutlery: z.boolean(),
+  napkins: z.boolean(),
+  base: z.object({
+    bread: z.union(literalMapper(breadVariants)),
+    cheese: z.array(z.union(literalMapper(cheeseVariants))),
+    meat: z.array(z.union(literalMapper(meatVariants))),
+    dressing: z.array(z.union(literalMapper(dressingVariants))),
+    vegetables: z.array(z.union(literalMapper(vegetableVariant))),
+  }),
+  extras: z.object({
+    egg: z.array(z.union(literalMapper(eggVariants))),
+    spreads: z.array(z.union(literalMapper(spreadVariant))),
+    serving: z.union(literalMapper(servingVariant)),
+    topping: z.union([z.literal(toppingVariant[0]), z.null()]),
+  }),
+});
+
+const SandwichDefaultVals: SandwichPayload = {
+  sandwichName: "Default Panini",
+  cutlery: false,
+  napkins: false,
+  base: {
+    bread: breadVariants[1] as "FULL GRAIN",
```

Comment:
**nitpick:** Such casting is typically a bad practice - you can try to make something like that:

```ts
const breadVariants = ["WHITE", "WHEAT", "FULL GRAIN"] as const;
const breadVariantsEnum = z.enum([...breadVariants]);

const SandwichDefaultVals: SandwichPayload = {
  ...
  base: {
    bread: breadVariants[2], // No need to cast, but the type of bread should be breadVariantsEnum
    ...
  },
  ...
};
```

Reply:
Fajna sprawa ten const :o